wear/src/main/kotlin/io/homeassistant/companion/android/home/HomeActivity.kt

```diff
@@ -55,7 +55,12 @@ class HomeActivity :
             }
         }
 
-        fun getLaunchAction(packageName: String, tileId: Int): ActionBuilders.LaunchAction {
+        sealed class LaunchMode(val mode: String) {
+            object ThermostatTile : LaunchMode(OpenTileSettingsActivity.CONFIG_THERMOSTAT_TILE)
+            object CameraTile : LaunchMode(OpenTileSettingsActivity.CONFIG_CAMERA_TILE)
+        }
```

Comment:
You don't need a `mode`

```suggestion
        sealed interface LaunchMode {
            object ThermostatTile : LaunchMode
            object CameraTile : LaunchMode
        }
```

When you do this then you can do something like 
```kotlin
when(launchMode) {
ThermostatTile -> // DO watever you want for thermostat
CameraTile -> // Same for Camera
```

With this pattern if you even add a new object the compiler is going to warn you at the `when` level that you need to handle the new case.

In your case maybe you want to create the sealed interface at the OpenTileSettingsActivity since the two options are for this activity not really for HomeActivity.

Reply:
Changed it: [a7aabbe](https://github.com/home-assistant/android/pull/5708/commits/a7aabbeae7121d020e3283c109b5797ff156ee39)

Reply:
Thanks. I looked a bit more and I've assumed wrong that it was possible to give an object to the `ActionBuilders` but it is not. So your proposal is good for me 👍🏻